async_producer.go

```diff
@@ -1227,6 +1231,19 @@ func (p *asyncProducer) retryHandler() {
 		}
 
 		buf.Add(msg)
+
+		if maxBufferSize > 0 && buf.Length() >= maxBufferSize {
```

Comment:
It may make sense to put a higher lower limit on the maxBuffer, since setting this to `1` would basically block all sending forever? This would see any non-zero maxBuffer but less than this lower limit to be push to the lower limit.

Reply:
@puellanivis your recommendation totally makes sense. I've added a const named `minFunctionalRetryBufferLength` at the top of this go file and used it in this `retryHandler` method.

In addition, I've modified the error message of `ErrProducerRetryBufferOverflow`, based on your comment. please take a look again. Thank you very much.